File: src/py/shuffle.py
from random import randrange

seq = [6, 1, 2, 4, 9, 3, 0, 5, 8, 7]

def shuffle(seq):
    for i in range(1, len(seq)):
        ran = randrange(i)
        seq[i], seq[ran] = seq[ran], seq[i]

shuffle(seq)
print seq 

File: src/js/shuffle.js
seq = [1, 2, 3, 4, 5, 6, 7, 8, 9];

function shuffle(seq) { 
    for(var i = 1; i < seq.length; i++){ 
        ran = getRandom(i);

        tmp = seq[i];
        seq[i] = seq[ran];
        seq[ran] = tmp;
    } 
}

function getRandom(n) {
    return Math.floor(Math.random() * n);
} 

shuffle(seq); 
for(var i = 1; i < seq.length; i++){
    console.log(seq[i]);
}
